internal/backend/sessions/sessionworkflows/workflow.go

```diff
@@ -251,10 +251,16 @@ func (w *sessionWorkflow) initGlobalModules() (map[string]sdktypes.Value, error)
 	return vs, nil
 }
 
-func (w *sessionWorkflow) createEventSubscription(ctx context.Context, connectionName, eventType string) (string, error) {
+func (w *sessionWorkflow) createEventSubscription(ctx context.Context, connectionName, filter string) (string, error) {
 	wctx := sessioncontext.GetWorkflowContext(ctx)
 	workflowID := workflow.GetInfo(wctx).WorkflowExecution.ID
-	signalID := fmt.Sprintf("wid_%s_cn_%s_et_%s", workflowID, connectionName, eventType)
+
+	var minSequence uint64
+	if err := workflow.ExecuteLocalActivity(wctx, w.ws.svcs.DB.GetLatestEventSequence).Get(wctx, &minSequence); err != nil {
+		w.z.Panic("get current sequence", zap.Error(err))
```

Comment:
I see you just moved this block from below, so it's not a deal-breaker, but I'm curious: since the `createEventSubscription` function can return errors, why do you convert errors from `ExecuteLocalActivity` to panics? Same question for the panics below in lines 275 and 280.

Reply:
you're right, these shouldn't be panics. i guess the intent was to make the workflow retry, but since the failures are activity failures, i think we should be good. fixed.